src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig

```diff
@@ -180,6 +180,11 @@
     </div>
 {%- endblock choice_widget_expanded %}
 
+{% block file_widget -%}
+    {%- set attr = attr|merge({class: (attr.class|default('') ~ ' custom-file-input')|trim}) -%}
+    {{- block('form_widget_simple', 'form_div_layout.html.twig') -}}
```

Comment:
You should remove 'form_div_layout.html.twig'. Or maybe Im missing something. Why did you add it here?

Reply:
I use form_div_layout.html.twig from form_widget_simple for overriding form_widget_simple from bootstrap_4_layout.html.twig.
Because we don't need merge .form-control-file with .custom-file-input.
I think we will need use only one class. And it will be .custom-file-input.
In <a href="https://getbootstrap.com/docs/4.0/components/forms/#file-browser">documention</a> used only .custom-file-input
Or am I wrong?

Reply:
Hm... Okey. Why are we adding `form-control-file`. Is that ever needed?

Reply:
https://getbootstrap.com/docs/4.0/components/input-group/#custom-file-input
In this example, we also not used form-control-file or form-control-label.

Reply:
Excuse my ignorance. But in what scenario/input do we need form-control-file?

Reply:
This block can be removed if you apply the solution specified before: edit the `form_simple_widget`. Your solution remove the way to use the previous possibilities to display file input, the only way become the custom way.

Then, just edit the part where we set the `.form-control-file` in place of `.form-control`, I've write the code to replace on the top. You just need to edit a line in the block `form_simple_widget` and remove this block.

I regive you the line of code:

In the `form_simple_widget` block there is:

    {%- set attr = attr|merge({class: (attr.class|default('') ~ ' form-control' ~ (type|default('') == 'file' ? '-file' : ''))|trim}) -%}

Replace by this line:

    {%- set attr = attr|merge({class: (attr.class|default('') ~ (type|default('') == 'file' ? 'custom-file-input' : 'form-control'))|trim}) -%}

Then, if the type is a file the `attr` become `custom-file-input` in place of `form-control`. The code is already ok, you just have to adapt it.

Reply:
We need form-control-file for adding: "display: block; width: 100%;". 
But, I am don't know if we need to use it in this case.

Reply:
In the demo they write it:

    <div class="input-group mb-3">
      <div class="custom-file">
        <input type="file" class="custom-file-input" id="inputGroupFile02">
        <label class="custom-file-label" for="inputGroupFile02">Choose file</label>
      </div>
    </div>

The input just have the class `custom-file-input`, this is what do the line:

    {%- set attr = attr|merge({class: (attr.class|default('') ~ (type|default('') == 'file' ? 'custom-file-input' : 'form-control'))|trim}) -%}